mesonpy/__init__.py

```diff
@@ -800,18 +784,11 @@ def _wheel_builder(self) -> _WheelBuilder:
             self._install_plan,
         )
 
-    def build_commands(self, install_dir: Optional[pathlib.Path] = None) -> Sequence[Sequence[str]]:
+    def build_commands(self) -> Sequence[Sequence[str]]:
         assert self._ninja is not None  # help mypy out
         return (
             (self._ninja, *self._meson_args['compile'],),
-            (
-                'meson',
-                'install',
-                '--only-changed',
-                '--destdir',
-                os.fspath(install_dir or self._install_dir),
-                *self._meson_args['install'],
-            ),
+            ('meson', 'install', '--no-rebuild', '--destdir', os.fspath(self._install_dir), *self._meson_args['install']),
```

Comment:
Why is `--only-changed` dropped here?

Reply:
It is explained in the commit message. `destdir` is always an empty temporary directory, thus `--only-changed` doesn't do anything at best, most likely it at at least requires an extra `stat()` system call for each file copied.

Reply:
I extended the commit message with some more background.

Reply:
Ah yes, the thing that changed is that this command is no longer run by an editable install. That's why `--only-changed` was needed before. Resolving this comment.